coredb-operator/charts/coredb-operator/templates/crd.yaml

```diff
@@ -93,6 +93,10 @@ spec:
                   image:
                     default: quay.io/prometheuscommunity/postgres-exporter:v0.12.0
                     type: string
+                  queries:
+                    x-kubernetes-preserve-unknown-fields: true
```

Comment:
Why do we need this?  I don't think I understand.

Reply:
We don't know the metric config names that come in queries.yaml, we don't know the keys ahead of time. e.g. the `pg_postmaster` in the yaml below - thats the "key", and that can be arbitrarily named. We don't know all the names it will be, so we can't set a struct attribute to "pg_postmaster". So in this case `pg_postmaster` is the unknown-field that we want to preserve.

```yaml
pg_postmaster:
  query: SELECT pg_postmaster_start_time as start_time_seconds from pg_postmaster_start_time()
  master: true
  metrics:
  - start_time_seconds:
      usage: Gauge
      description: Time at which postmaster started
 ```

Reply:
without it we get
```
strict decoding error: unknown field "spec.metrics.queries. pg_postmaster"
```

Reply:
Ah ok. Thanks for the explanation, that is correct then.